src/memory_targets/memory_target_numa.c

```diff
@@ -195,12 +195,47 @@ static umf_result_t numa_get_capacity(void *memTarget, size_t *capacity) {
     return UMF_RESULT_SUCCESS;
 }
 
+static umf_result_t numa_get_bandwidth(void *memTarget, size_t *bandwidth) {
+    hwloc_topology_t topology = umfGetTopology();
+    if (!topology) {
+        return UMF_RESULT_ERROR_NOT_SUPPORTED;
+    }
+
+    hwloc_obj_t numaNode = hwloc_get_obj_by_type(
+        topology, HWLOC_OBJ_NUMANODE,
+        ((struct numa_memory_target_t *)memTarget)->physical_id);
+    if (!numaNode) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    umf_result_t umf_ret = UMF_RESULT_SUCCESS;
+
+    hwloc_const_cpuset_t cCpuset = hwloc_topology_get_allowed_cpuset(topology);
+    hwloc_cpuset_t cpuset = hwloc_bitmap_dup(cCpuset);
+    struct hwloc_location initiator = {.location.cpuset = cpuset,
+                                       .type = HWLOC_LOCATION_TYPE_CPUSET};
+    hwloc_uint64_t value = 0;
+    int ret = hwloc_memattr_get_value(topology, HWLOC_MEMATTR_ID_BANDWIDTH,
```

Comment:
What will this function return if there are multiple nodes in the cpuset? Shouldn't `numa_get_bandwidth` accept an initiator node? We could then look at bandwidth between every pair of nodes in init code. I think we do that in memkind already:

https://github.com/memkind/memkind/blob/c425635feb0367c43da8fa0deadb50a4bc127988/src/memkind_mem_attributes.c#L324

Reply:
`numa_get_bandwidth` accepts a `memTarget` which represents a node, in this case the initiator is the cpuset on the platform. Do you mean that we should get the `cpuset` information from the HWLOC numa node obj structure?

Reply:
I was suggesting that perhaps we should accept both `initiator` and `target` nodes as params in `numa_get_bandwidth` because we are interested in bandwidth between every pair of nodes I think.